doc/members/proposals.rst

```diff
@@ -67,6 +67,24 @@ Some of these subcommands require additional arguments, such as the node ID or u
 
 These proposals and votes should be sent as the body of HTTP requests as described below.
 
+Creating Proposals in Python
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``ccf.proposal_generator`` can also be imported and used in a Python application instead of as a command-line tool.
+
+.. code-block:: python
+
+    import ccf.proposal_generator
+
+    proposal, vote = ccf.proposal_generator.open_network()
+
+    # Suppose `ccf_client` is an instance of `CCFClient` with 
+    # the default `RequestClient` client implementation.
+    response = ccf_client.post(
+        "/gov/proposals", body=proposal, signed=True,
+    )
+
+Avoid using the ``ccf.proposal_generator.ProposalGenerator`` class, which wraps the proposal generation functions to write proposals and votes to disk as JSON instead of directly returning dictionaries.
```

Comment:
This isn't quite right. The `ProposalGenerator` class writes to file, but still returns the useable proposal. If it writes `proposal_foo.json`, it returns the string `"@proposal_foo.json"`, which both the `RequestClient` and `CurlClient` will treat curl-style (ie - use the contents of the path). This is how our tests generate proposals. The advantage is that the proposals can be inspected after the fact. So if the caller wants to inspect or modify the proposal as a dict, they can use `ccf.proposal_generator`, but in most cases they should also be able to use a `ccf.proposal_generator.ProposalGenerator` instance.

Reply:
That's my fault - I didn't read through `RequestClient` exactly and assumed it wouldn't work with the `@<filename>` input since `requests` calls take a `dict`. I thought the `ProposalGenerator` was just for the `CurlClient`.